modules/my_health/app/controllers/my_health/v1/prescriptions_controller.rb

```diff
@@ -51,13 +51,20 @@ def collection_resource
       end
 
       def last_refill_date_filter(resource)
-        resource = resource.sort_by do |x|
-          x.rx_rf_records?.rf_record.find do |rf_record|
-            rf_record[:dispensed_date].present?
-          end.present? || x[:dispensed_date].present?
-        end
+        resource = resource.data.sort_by do |x|
+          has_refills = x.try(:rx_rf_records).present?
+
+          last_refill_date = 30.years.ago
+          last_refill_date = x.rx_rf_records.flatten[1][:dispensed_date]  if has_refills
+
+          if last_refill_date > 30.years.ago
+            last_refill_date.to_date.strftime('%s').to_i
+          else
+            x.dispensed_date.present? ? x.dispensed_date.to_date.strftime('%s').to_i : DateTime.new(1999).to_i
```

Comment:
What's the significance of 1999 here? Will it need to be updated in the future?

Reply:
Some prescriptions don't have a `dispensed_date` -- I chose an arbitrary old number so prescriptions with a `nil` value will appear at the bottom.

Reply:
gotcha - would it make sense to just use 0?